%% BioMed_Central_Tex_Template_v1.05
%%                                      %
%  bmc_article.tex            ver: 1.05 %
%                                       %


%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
%%                                     %%
%%  LaTeX template for BioMed Central  %%
%%     journal article submissions     %%
%%                                     %%
%%         <27 January 2006>           %%
%%                                     %%
%%                                     %%
%% Uses:                               %%
%% cite.sty, url.sty, bmc_article.cls  %%
%% ifthen.sty. multicol.sty            %%
%%                                     %%
%%                                     %%
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%


%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
%%                                                                 %%	
%% For instructions on how to fill out this Tex template           %%
%% document please refer to Readme.pdf and the instructions for    %%
%% authors page on the biomed central website                      %%
%% http://www.biomedcentral.com/info/authors/                      %%
%%                                                                 %%
%% Please do not use \input{...} to include other tex files.       %%
%% Submit your LaTeX manuscript as one .tex document.              %%
%%                                                                 %%
%% All additional figures and files should be attached             %%
%% separately and not embedded in the \TeX\ document itself.       %%
%%                                                                 %%
%% BioMed Central currently use the MikTex distribution of         %%
%% TeX for Windows) of TeX and LaTeX.  This is available from      %%
%% http://www.miktex.org                                           %%
%%                                                                 %%
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%


\NeedsTeXFormat{LaTeX2e}[1995/12/01]
\documentclass[10pt]{bmc_article}    



% Load packages
\usepackage{cite} % Make references as [1-4], not [1,2,3,4]
\usepackage{url}  % Formatting web addresses  
\usepackage{ifthen}  % Conditional 
\usepackage{multicol}   %Columns
\usepackage[utf8]{inputenc} %unicode support
%\usepackage[applemac]{inputenc} %applemac support if unicode package fails
%\usepackage[latin1]{inputenc} %UNIX support if unicode package fails
\usepackage{siunitx}
\usepackage{etoolbox}
\usepackage{booktabs}
\usepackage{xspace}
\urlstyle{rm}
 
 
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%	
%%                                             %%
%%  If you wish to display your graphics for   %%
%%  your own use using includegraphic or       %%
%%  includegraphics, then comment out the      %%
%%  following two lines of code.               %%   
%%  NB: These line *must* be included when     %%
%%  submitting to BMC.                         %% 
%%  All figure files must be submitted as      %%
%%  separate graphics through the BMC          %%
%%  submission process, not included in the    %% 
%%  submitted article.                         %% 
%%                                             %%
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%                     


\def\includegraphic{}
\def\includegraphics{}



\setlength{\topmargin}{0.0cm}
\setlength{\textheight}{21.5cm}
\setlength{\oddsidemargin}{0cm} 
\setlength{\textwidth}{16.5cm}
\setlength{\columnsep}{0.6cm}

\newboolean{publ}

%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
%%                                              %%
%% You may change the following style settings  %%
%% Should you wish to format your article       %%
%% in a publication style for printing out and  %%
%% sharing with colleagues, but ensure that     %%
%% before submitting to BMC that the style is   %%
%% returned to the Review style setting.        %%
%%                                              %%
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
 

%Review style settings
%\newenvironment{bmcformat}{\begin{raggedright}\baselineskip20pt\sloppy\setboolean{publ}{false}}{\end{raggedright}\baselineskip20pt\sloppy}

%Publication style settings
\newenvironment{bmcformat}{\fussy\setboolean{publ}{true}}{\fussy}



% Begin ...
\begin{document}
\begin{bmcformat}


%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
%%                                          %%
%% Enter the title of your article here     %%
%%                                          %%
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
\newcommand{\quorum}{QuorUM\xspace}

\title{\quorum: an error corrector for Illumina reads}
 
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
%%                                          %%
%% Enter the authors here                   %%
%%                                          %%
%% Ensure \and is entered between all but   %%
%% the last two authors. This will be       %%
%% replaced by a comma in the final article %%
%%                                          %%
%% Ensure there are no trailing spaces at   %% 
%% the ends of the lines                    %%     	
%%                                          %%
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%


\author{Guillaume Mar\c{c}ais\correspondingauthor$^1$%
       \email{Guillaume Mar\c{c}ais\correspondingauthor - gmarcais@umd.edu}%
      \and
         Aleksey Zimin\correspondingauthor$^1$%
         \email{Aleksey Zimin\correspondingauthor- alekseyz@ipst.umd.edu}%
       \ and
         James A. Yorke$^1$%
         \email{James Yorke - yorke@umd.edu}
      }
      

%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
%%                                          %%
%% Enter the authors' addresses here        %%
%%                                          %%
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%

\address{%
    \iid(1)University of Maryland, College Park, MD
}%

\maketitle

%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
%%                                          %%
%% The Abstract begins here                 %%
%%                                          %%
%% The Section headings here are those for  %%
%% a Research article submitted to a        %%
%% BMC-Series journal.                      %%  
%%                                          %%
%% If your article is not of this type,     %%
%% then refer to the Instructions for       %%
%% authors on http://www.biomedcentral.com  %%
%% and change the section headings          %%
%% accordingly.                             %%   
%%                                          %%
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%


\begin{abstract}
  % Do not use inserted blank lines (ie \\) until main body of text.
\paragraph*{Background:}
Illumina Sequencing data can provide high coverage of a genome by relatively short (\SI{100}{bp} to \SI{150}{bp}) reads at a low cost.
The base error rates in the reads vary greatly along the read sequence.
Low quality bases can be trimmed off or error-corrected.
Trimming, especially based on the quality scores, can eliminate large amounts of useful sequence in the reads.
Error correction is an alternative to trimming that takes advantage of the high coverage of the genome to make the high confidence base corrections in the reads.
Error correction allows for more effective use of the sequencing data thus reducing the coverage depth requirements in sequencing projects and increasing the quality of the resulting assemblies.
The vast majority of errors in Illumina data are substitutions, where a base call is incorrect, and \quorum focuses on this type of error.
      
\paragraph*{Results:}
We propose software, called \quorum, for error correcting substitution error in Illumina reads.
It provides accurate correction and is suitable for large data sets ($48$ billion bases corrected per hours on $48$ threads).

\paragraph*{Conclusions:} 
The software is available under the GPL open source license at \url{ftp://ftp.genome.umd.edu/pub/quorum}.

\end{abstract}



\ifthenelse{\boolean{publ}}{\begin{multicols}{2}}{}




%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
%%                                          %%
%% The Main Body begins here                %%
%%                                          %%
%% The Section headings here are those for  %%
%% a Research article submitted to a        %%
%% BMC-Series journal.                      %%  
%%                                          %%
%% If your article is not of this type,     %%
%% then refer to the instructions for       %%
%% authors on:                              %%
%% http://www.biomedcentral.com/info/authors%%
%% and change the section headings          %%
%% accordingly.                             %% 
%%                                          %%
%% See the Results and Discussion section   %%
%% for details on how to create sub-sections%%
%%                                          %%
%% use \cite{...} to cite references        %%
%%  \cite{koon} and                         %%
%%  \cite{oreg,khar,zvai,xjon,schn,pond}    %%
%%  \nocite{smith,marg,hunn,advi,koha,mouse}%%
%%                                          %%  
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%




%%%%%%%%%%%%%%%%
%% Background %%
%%
\section*{Background}
While second generation sequencing technologies have progressed tremendously and offer ever longer reads with low overall sequencing error rate, correcting errors in reads remains an important prepossessing step in \emph{de novo} genome assembly.
In general, error correcting the reads leads to assemblies with longer contiguous sequences and fewer misassemblies.


As the quality of base calling usually degrades toward the 3' ends of reads, an obvious error correction method is to trim all the reads on the 3' end, either by a fixed amount or based on the quality values reported by the sequencing machine.
Although this simple trimming scheme will reduce the number of erroneous bases, it still leaves many errors in the reads and needlessly discards a lot of valid sequence. 
% G: You should add a trimmer that trims at 5 percent
Aggressive trimming results in fragmented assemblies.

On the other hand, trimming can be an integral part of error correction.
The distribution of sequencing errors on the reads is complex and for some percentage of the reads, the sequence beyond a certain point contains too many errors to be corrected or, even worse, does not correspond to any sequence in the original genome.
It is important to trim those reads to avoid misassemblies.
See [[Gage Reference]].

% We need a paragraph or two on the various way to error correct: k-mer spectrum, alignment of reads, what else?
Genomes are sequenced at some depth.
Meaning that every part of the genome should be present in more than one sequencing read.
As sequencing machine make a mistake reading a base, this error can be detected by comparing the read to others covering the same region of the genome.
Put another way, erroneous base should follow a different probability distribution that correct bases.

Existing error correctors use different techniques to detect and correct these erroneous bases.
Coral~\cite{salmela2011correcting} and Echo~\cite{Kao2011} use multiple alignment of the reads.
Then, a statistical model of sequencing is used to correct misaligned bases.
HiTec~\cite{Ilie2011b} uses a suffix array to find and correct potentially erroneous bases.
Quake~\cite{Kelley:2010fk} uses the distribution of $k$-mers, or the $k$-spectrum, to detect $k$-mers containing erroneous bases.

We propose a new error correction procedure and software package, named \quorum (Quality Optimized Reads from the University of Maryland), that provides accurate trimming and error correction.
\quorum uses both the $k$-spectrum and the quality scores reported by the sequencing machine.
In addition, \quorum is flexible: it can correct reads of varying length and reads containing ambiguous base calls (Ns).
% TODOcheck number below
It is also fast (1 billion $\SI{100}{bases}$ long reads in a day) and can tackle the large data sets produced by today's high throughput sequencing machine.

For evaluation we investigate two genomes that have Illumina reads and have genome assemblies of finished quality.
We compare the error correctors by aligning the error corrected reads to the finished sequence of the organism and using the following metrics.

% First, we report the number of substitution errors left after error correction and the number of new errors introduced (i.e.\@ false positive rate) (Table~\ref{table:left-introduced}).
% This measures how well the software corrects individual substitution errors.
% %
% Second, we measure the number of chimeric reads before and after error correction (Table~\ref{table:chimeric}).
% % Def chimeric has been added. Is this OK?
% A read is called chimeric if it does not match the finished genome in one piece but the read can be split into two parts that match the genome in two widely separated places.
% This measures the likelihood  of the software to over correct by gluing sequences that do not belong together.
% %
% Third, we report the amount of sequence which is error free reads (i.e.\@ reads which aligns perfectly to the finished sequence) (Table~\ref{table:perfect-reads}).
% These reads are easy for assemblers to use, and this is a measure of how well the correction and trimming work.
% %
% Fourth, we report the contig sizes of ``idealized contigs'' created by aligning error corrected reads to the finished sequence (Table~\ref{table:icontigs}).
% This procedure, similar to a guided assembly, provides some estimate of the maximum contig size achievable through assembly, without the bias of using a particular assembler.
% Finally, we will report the time and memory consumption to carry out the correction.

% We use the evaluation toolkit of~\cite{Yang2012} with some modifications (see~\ref{sec:Methods}).

 
%%%%%%%%%%%%%%%%%%%%%%%%%%%%
%% Results and Discussion %%
%%
\section*{Results and Discussion}

We evaluated the error correction software by error correcting the reads of two organisms for which a finished sequence is available: \emph{Rhodobacter sphareoides} and \emph{Mus musculus} (house mouse).
These genomes present different type of challenges for error correction.
The \emph{Rhodobacter sphaeroides} bacteria genome (\SI{4.6}{Mb} long) has a high GC content and is consequently difficult to sequence using Illumina technology.
The mouse genome, being a mammalian genome, is larger and more complex.
For the sake of simplicity, we only use chromosome 16 of the mouse genome which has \SI{98}{Mb} in finished sequence.

We ran the following error correctors on the Rhodobacter data set: Echo, HiTEC, Coral, \quorum, and Quake.
Echo and HiTEC did not run on the larger mouse data.

As can be seen in Table~\ref{table:perfect-reads}, \quorum produces the largest amount of sequence in perfect reads.
And these perfect reads represent a large proportion of the error corrected reads ($94.3\%$ in the mouse and $98.7\%$ in the Rhodobacter).
Although \quorum trims and discards some reads (see Table~\ref{table:left-introduced} and \ref{table:perfect-reads}), the amount of perfect sequence is larger than that produce by the correctors that do not trim or discard any read, such as Echo, HiTEC and Coral.
Moreover, \quorum trims less aggressively than Quake while still producing a high percentage of perfect reads.

When comparing the error corrected reads to the original reads (see Table~\ref{table:left-introduced}), \quorum outputs reads with the fewest errors while introducing only a small fraction of new errors (False Positive Rate).
% In the current version of the paper, we cannot tell how many errors are left. We can only tell how many are left in the trimmed region. It might be getting 99 percent of the errors, for all we know.
Quake has a smaller false positive rate but at the cost of leaving many errors, despite trimming the reads more aggressively. 

The Illumina technology generates few chimeric reads (usually less than $1\%$ of all the reads).
When doing very aggressive error correction, one runs the risk of creating new chimeric reads;
i.e.\@ sequence from a distant region of the genome is used to rewrite significant portion of a read.
Table~\ref{table:chimeric} shows that Quake and \quorum decrease significantly the number of chimeric reads.
%Echo, HiTEC and Coral do not change or 
HiTEC actually  increases the number of chimeric reads while while Echo and Coral make only small decreases in chimeric reads.\\

Finally, Table~\ref{table:chimeric} shows that reads error corrected by \quorum leads to larger contigs than with the other error correctors.


%%%%%%%%%%%%%%%%%%%%%%
\section*{Conclusions}


  
%%%%%%%%%%%%%%%%%%
\section*{Methods}
\label{sec:Methods}

\subsection*{Error correction}
\label{sec:meth-error-correction}

\quorum error corrects reads by replacing low quality $k$-mers in the input reads with high quality $k$-mers.
The quality of a $k$-mer is based first on the quality values reported by the sequencing machine of the bases in the mer and second by its number of occurrences in the reads.

\quorum uses the Jellyfish~\cite{Marcais2011} software to count the number of occurrences of the $k$-mers in the input reads.
First it counts the occurrences of all $k$-mers and second it counts the occurrences $k$-mers in which all bases have a quality value greater than a threshold (by default $5$ above the minimum quality). We call the latter $k$-mers ``high quality''.
%This second set of $k$-mers is the set of high-quality $k$-mers.
Then, every read is examined in turn.
\quorum searches for an \emph{anchor} $k$-mer: a $k$-mer whose number of occurrences is higher than a threshold $a$ ($a = 3$ by default).
\quorum searches for an \emph{anchor} $k$-mer: a $k$-mer whose number of occurrences is higher than a threshold $a$ ($a = 3$ by default). 
Starting from this anchor, \quorum will look at every base, extending toward the 5' and 3' ends, and decide to make a correction based on the $k$-mer ending at that base.
Let $m = xb$ bet the $k$-mer, where $|m| = k$, $x$ is a prefix with $|x| = k - 1$, and $b$ is the base under consideration.
If $m$ has a number of occurrences greater than $a$, then no correction is made.
Otherwise, the 3 possible alternatives $b'$ are considered.
\quorum considers in priority the $k$-mers $m' = xb'$ in the high-quality set.
Only if all $m'$ have count 0 in the high-quality set is the set containing all $k$-mers considered.
\quorum substitutes base $b$ by $b'$ if only one of the alternative has a occurrence count at least 3 times that of $m$.
Moreover, \quorum checks that this substitution is not a dead-end.
I.e.\@ that at least one the of four extensions of $m'$ is present in the input reads, where the extensions of $m'$ are the four possible $k$-suffixes of $m'b''$ and $b''$ is a base.

{\bf Correcting a base in a read.}
Starting from this anchor, \quorum will look at every base, extending toward the 5' and 3' ends.  Here we consider extension toward the 3' end and leave the reverse to the reader. As \quorum shifts the k-mer toward the 3' end on base at a time, it and decides whether 
% [[]]
to make a correction based on the $k$-mer ending at that base.
Let $m = xb$ be the anchor $k$-mer, where the length $|m| = k$, and $x$ is a prefix with length $|x| = k - 1$, and $b$ is the base under consideration.
If $m$ has a number of occurrences greater than $a$ in the entire database, then no correction is made; otherwise, the 3 possible alternative bases $b'$ are considered.
\quorum determines if any of the three $k$-mers $m' = xb'$ are in the high-quality set.
Only if all three $m'$ have count 0 in the high-quality set is the set containing all $k$-mers considered. Then
\quorum substitutes base $b'$ for $b$ if only one of the alternative $xb'$ considered has an occurrence count at least 3 times that of $m$.
Moreover, \quorum checks that this substitution is not a dead-end; i.e.\@ at least one the of four extensions of $m'$ is present in the input reads, where the extensions of $m'$ are the four possible $k$-suffixes of $m'b''$ and $b''$ is a base. If there were two base errors in a row, \quorum would be very unlikely to make a change. 

The maximum number of corrections made in a window of length $10$ is by default capped at $3$.
\quorum will trim the read at the beginning of the window if the number of errors is higher.
% Jim: Is this last sentence correct?
% Gus: Not sure what it means
The sequence in $m$ and beyond to the 5' end is trimmed off the read. 

% These options were not used in the work on this paper. Yes or No???
% The first option was not used. The second one was the Rhodobacter data.
In addition, \quorum has two extra optional procedure for trimming reads. 
It takes as input a set of contaminant $k$-mers and it will trim a read wherever it contains such a contaminant mer. In particular the contaminant is deleted.

Also, in high or low GC genome, the Illumina sequencers tend to erroneously generate long homo-polymer run toward the 3' end of the reads.
\quorum can optionally detect and trim off these homo-polymer.


\subsection*{Evaluation}

Need to add the access IDs of the reads and finished sequence use. Add some statistics: number of reads, coverage, size of genomes.
Mention the aligner used and the parameter used.

Couple of sentences on the benefit of using finished quality genome with real sequencing reads (not artificially sequenced).
    
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
\section*{Authors contributions}


    

%%%%%%%%%%%%%%%%%%%%%%%%%%%
\section*{Acknowledgements}
  \ifthenelse{\boolean{publ}}{\small}{}
  Text for this section \ldots


 
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
%%                  The Bibliography                       %%
%%                                                         %%              
%%  Bmc_article.bst  will be used to                       %%
%%  create a .BBL file for submission, which includes      %%
%%  XML structured for BMC.                                %%
%%                                                         %%
%%                                                         %%
%%  Note that the displayed Bibliography will not          %% 
%%  necessarily be rendered by Latex exactly as specified  %%
%%  in the online Instructions for Authors.                %% 
%%                                                         %%
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%


{
  \ifthenelse{\boolean{publ}}{\footnotesize}{\small}
  \bibliographystyle{bmc_article}  % Style BST file
  \bibliography{bmc_article}
}     % Bibliography file (usually '*.bib' ) 

%%%%%%%%%%%

\ifthenelse{\boolean{publ}}{\end{multicols}}{}

%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
%%                               %%
%% Figures                       %%
%%                               %%
%% NB: this is for captions and  %%
%% Titles. All graphics must be  %%
%% submitted separately and NOT  %%
%% included in the Tex document  %%
%%                               %%
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%

%%
%% Do not use \listoffigures as most will included as separate files

% \section*{Figures}
% \subsection*{Figure 1 - Sample figure title}
% A short description of the figure content
% should go here.

% \subsection*{Figure 2 - Sample figure title}
% Figure legend text.



%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
%%                               %%
%% Tables                        %%
%%                               %%
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%

% \newrobustcmd{\bnum}[1]{\bfseries #1}

%% Use of \listoftables is discouraged.
%%
\newcounter{tablecounter}
\newcommand{\tabledesc}[1]{%
\refstepcounter{tablecounter}%
\subsection*{Table \thetablecounter\  - #1}%
}
%% Winner number in a table
\newcommand{\winner}[1]{\bfseries #1}
\newcommand{\tn}[1]{S[detect-weight,table-format=#1]}

\section*{Tables}
\robustify\bfseries

%
%
% \tabledesc{Mapping contig lengths}
% \label{table:icontigs}

% We map the corrected reads onto the finished sequence.
% We require $\ge 99\%$ agreement; that is, since the reads are $101$ bases long, we allow at most one error reads that retain full length after correcting, and perfect matches for trimmed reads.
% We allow each read to be mapped to at most $5$ distinct locations.
% Each contiguous segment of the genome that is covered by overlapping reads is called a ``mapping contig''.
% Two consecutive mapping contigs are separated by a gap in coverage.
% This table shows the statistics on the mapping contig lengths (N25, N50 and N90) in kilo bases.

% %% TODO move following to text
% Trimming inevitably discards some correct sequence.
% There is a trade-off between trying to correct errors and trimming off hard-to-correct ends of a read.
% We note that the contig sizes are reduced when reads are trimmed and when errors remain.
% A trimmed read covers fewer bases and aggressive trimming will unnecessarily shorten reads and mapping contigs.
% On the other hand, a read may fail to align because of untrimmed errors.


% \bigskip

% % Table 1: E values look wrong for the mouse. Recompute.

% % Table 1 as is goes to supplementary material. New table 1 is for
% % perfect reads only; N50, E and percentage of perfect reads in output reads

% % Quake throws away lots of read but does an excellent job on what it
% % outputs. It seems likely that this deficiency could be compensated
% % by extra sequencing.

% \begin{tabular}{@{}lS[detect-weight,table-format=2.2]S[detect-weight,table-format=2.2]S[detect-weight,table-format=1.3]S[detect-weight,table-format=2.2]cS[detect-weight,table-format=3.1]S[detect-weight,table-format=2.1]S[detect-weight,table-format=2.2]S[detect-weight,table-format=2.1]@{}}
% \toprule
% Corrector & \multicolumn{4}{c}{Rhodobacter}                              & & \multicolumn{4}{c}{Mouse} \\
% \cmidrule(r){2-5} \cmidrule(l){7-10}
% {}        & {N25 kb}      & {N50 kb}      & {N90 kb}      & {E kb}       & &  {N25 kb}     & {N50 kb}      & {N90 kb}      & {E kb}        \\
% \midrule                                                                     
% none      & 4.77          & 2.56          & 0.614         & 3.99         & &  62.8         & 35.7          & 8.58          & \winner{26.4} \\
% trim20B   & 13.4          & 7.88          & 2.42          & 10.8         & &  98.2         & 59.5          & 14.5          & 25.6          \\
% trimQual5 & 10.9          & 5.88          & 1.73          & 7.87         & &  \winner{105} & 62.3          & 14.8          & 25.3          \\
% coral     & 10.8          & 6.10          & 1.61          & 8.50         & &  32.6         & 18.3          & 2.81          & 20.2          \\
% echo      & 12.6          & 6.96          & 1.86          & 9.00         & &               &               &               &               \\
% hitec     & 11.3          & 6.56          & 1.80          & 9.19         & &               &               &               &               \\
% quake     & 11.2          & 6.50          & 1.81          & 9.41         & &  50.2         & 29.5          & 7.08          & 25.9          \\
% quorum    & \winner{38.2} & \winner{23.8} & \winner{7.02} & \winner{24.1}& &  104          & \winner{62.8} & \winner{14.9} & 26.0          \\
% \bottomrule
% \end{tabular}

% \bigskip

% Original data
% Mouse:
% \begin{tabular}{|c|c|c|c|c|}
% corrector     & N25     & N50    & N90    & E      \\
% identity      & 62,758  & 35,735 & 8,581  & 26,350 \\
% trimmer       & 98,222  & 59,540 & 14,452 & 25,603 \\
% qual\_trimmer & 104,540 & 62,252 & 14,778 & 25,289 \\
% coral         & 32,580  & 18,311 & 2,807  & 20,220 \\
% quorum        & 104,357 & 62,837 & 14,926 & 26,049 \\
% quake         & 93,219  & 53,358 & 12,925 & 26,155 \\
% quake2        & 50,164  & 29,520 & 7,084  & 25,942 \\
% \end{tabular}

% \bigskip

% Rhodobacter:
% \begin{tabular}{|c|c|c|c|c|}
% corrector     & N25   & N50   & N90  & E       \\
% identity      & 4767  & 2563  & 614  & 3993.29 \\
% trimmer       & 13413 & 7881  & 2415 & 10833.2 \\
% qual\_trimmer & 10926 & 5881  & 1731 & 7870.86 \\
% quorum        & 38190 & 23829 & 7021 & 24082.7 \\
% echo          & 12567 & 6957  & 1861 & 8996.44 \\
% hitec         & 11292 & 6557  & 1795 & 9190.29 \\
% quakedk       & 11241 & 6504  & 1809 & 9409.1  \\
% coral         & 10801 & 6096  & 1610 & 8497.04 \\
% \end{tabular}

%
%
\tabledesc{Errors in output including introduced errors}
\label{table:left-introduced}

The ``Bases trimmed'' is the percentage of bases in trimmed reads compared with bases in the original reads.
All the bases in a discarded read count as bases trimmed off.
The ``Errors not trimmed from reads'' is the percentage of errors in the trimmed reads compared with the errors in the original reads.
The ``errors left in trimmed reads'' is the percentage of errors in the corrected reads compared with the errors in the trimmed reads.
The ``Errors introduced'' is the number of errors introduced divided by the number of errors in the original reads.
These are counted in the previous column ``errors left in trimmed reads''.
Finally, the ``Errors in corrected trimmed reads'' is the percentage of errors in corrected reads compared with the errors in the original reads.
If we name the four ``errors'' columns as $E1$, $E2$, $E3$, and $E4$, then $E1 * E2 = E4$.

The two correctors that both trim and error correct what is left do the two steps of trimming and correcting concurrently.

\bigskip

\begin{tabular}{@{}lS[detect-weight,table-format=2.1]S[detect-weight,table-format=3.1]S[detect-weight,table-format=2.2]S[detect-weight,table-format=2.1]S[detect-weight,table-format=2.1]@{}}
\toprule
{}        & {\% bases} & {\% errors not}      & {\% of errors left} & {\% errors}    & {\% errors in corrected} \\
Corrector & {trimmed}  & {trimmed from reads} & {in trimmed reads}  & {introduced}   & {trimmed reads}          \\
\midrule
{}        & \multicolumn{5}{c}{Rhodobacter}                                                                     \\
\cmidrule{2-6} 
trim20B   & 19.8       & 43.9                 & 100                 & 0.00           & 43.9                     \\
trimQual5 & 36.0       & 11.2                 & 100                 & 0.00           & 11.2                     \\
coral     & 0.00       & 100                  & 77.7                & 0.766          & 77.7                     \\
echo      & 0.00       & 100                  & 65.1                & 2.11           & 65.1                     \\
hitec     & 0.00       & 100                  & 61.1                & 10.4           & 61.1                     \\
quake     & 30.6       & 81.5                 & 98.2                & 0.365          & 80.1                     \\ 
quorum    & 31.0       & \winner{21.7}        & \winner{10.8}       & \winner{.0700} & \winner{2.33}            \\
\cmidrule{2-6}
{}        & \multicolumn{5}{c}{Mouse}                                                                           \\
\cmidrule{2-6}
trim20B   & 19.8       & 29.0                 & 100                 & 0.00           & 29.0                     \\
trimQual5 & 10.5       & 28.0                 & 100                 & 0.00           & 28.0                     \\
coral     & 0.00       & 100                  & 69.0                & 12.5           & 69.0                     \\
quake     & 14.2       & \winner{10.4}        & 71.5                & \winner{.494}  & \winner{7.40}            \\
quorum    & 6.75       & 29.4                 & \winner{34.7}       & .786           & 10.2                     \\
\cmidrule{2-6}
{}        & \multicolumn{5}{c}{Staphylococcus}                                                                  \\
\cmidrule{2-6}
trim20B   & 19.8       & 53.8                 & 100                 & 0.00           & 53.8                     \\
trimQual5 & 1.30       & 91.7                 & 100                 & 0.00           & 91.7                     \\
coral     & 0.00       & 100                  & 52.8                & 1.08           & 52.8                     \\
echo      & 0.00       & 100                  & 38.6                & .628           & .386                     \\
hitec     & 0.00       & 100                  & 21.3                & 1.57           & 21.3                     \\
quake     & 42.0       & \winner{13 .7}       & \winner{4.98}       & \winner{.115}  & \winner{.682}            \\
quorum    & 18.3       & 55.1                 & 7.38                & .271           & 4.07                     \\
\bottomrule
\end{tabular}

%
%
\tabledesc{Chimeric reads}

\label{table:chimeric}

% Jim: We have to make sure that our definitions agree with what I put in the main text. Two distinct pieces allows one to be extremely small and it does not say that the whole read is the union of the two pieces.
A chimeric read is defined here as a read that does not align in one piece while it has two disjoint subsequences that align to far apart regions of the finished sequence. 
The table shows the number of chimeric reads after error correction. ``None'' denote the number without any error correction.

\bigskip

\begin{tabular}{@{}lS[detect-weight,table-format=4]S[detect-weight,table-format=6]S[detect-weight,table-format=4]@{}}
\toprule
{Corrector} & {Rhodobacter} & {Mouse}        & {Staphylococcus} \\
\midrule                                                   
none        & 2086          & 296772         & 733              \\
trim20B     & 1655          & 250382         & 597              \\
trimQual5   & 649           & 148439         & 720              \\
coral       & 2047          & 225052         & 1024             \\
echo        & 1809          &                & 704              \\
hitec       & 6375          &                & 1200             \\
quake       & 1007          & \winner{63079} & 651              \\
quorum      & \winner{490}  & 93158          & \winner{514}     \\
\bottomrule
\end{tabular}

%
%
\tabledesc{Output bases in perfect reads \& percent of output bases in perfect reads}
\label{table:perfect-reads}

The columns labeled ``Perfect'' show the sum of the lengths of the perfect reads after error correction.
A perfect read is a read (possibly trimmed) that aligns with no error on its full length to the finish sequence.
The columns labeled ``\%'' show the number of bases in perfect corrected reads over the number of bases in corrected reads, shown as a percentage.

Hitec and Echo did not run on the mouse data set within the 24 hour time frame.

% Number for coral worse than none seems bad. Check this number. Justification to add: it collapses repeats, hence it creates gaps.

% Check the number in this table: do not seem to match definition. Add numbers for the trimmers.
% The number we care about is the percentage of perfect reads in the output reads (not amount of sequence).

\bigskip

\begin{tabular}{@{}lS[detect-weight,table-format=2.1]S[detect-weight,table-format=2.1]S[detect-weight,table-format=1.2]S[detect-weight,table-format=2.1]S[detect-weight,table-format=2.1]S[detect-weight,table-format=2.1]@{}}
\toprule
Corrector & \multicolumn{2}{c}{Rhodobacter} & \multicolumn{2}{c}{Mouse} &\multicolumn{2}{c}{Staphylococcus}                      \\
\cmidrule(r){2-3} \cmidrule(l){4-5} \cmidrule(l){6-7}
{}        & {Reads (\%)}  & {Sequence (\%)} & {Reads (\%)}  & {Sequence (\%)} & {Reads (\%)}  & {Sequence (\%)} \\
\midrule
none      & 20.9          & 20.9            & 48.3          & 48.3            & 33.1          & 33.1            \\
trim20B   & 44.4          & 35.6            & 79.4          & 63.6            & 46.3          & 37.2            \\
trimQual5 & 76.2          & 51.2            & 78.0          & 71.9            & 35.3          & 35.1            \\
coral     & 58.2          & 58.2            & 80.9          & 80.9            & 73.5          & 73.5            \\
echo      & 56.3          & 56.3            &               &                 & 65.1          & 65.1            \\
hitec     & 60.8          & 60.8            &               &                 & 77.9          & 77.9            \\
quake     & 61.9          & 48.5            & 88.7          & 81.3            & 68.2          & 57.6            \\
\quorum   & \winner{88.8} & \winner{77.1}   & \winner{90.9} & \winner{86.9}   & \winner{81.9} & \winner{78.4}   \\
\bottomrule
\end{tabular}

%
%


%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
%%                               %%
%% Additional Files              %%
%%                               %%
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%

% \section*{Additional Files}
% \subsection*{Additional file 1 --- Sample additional file title}
% Additional file descriptions text (including details of how to
% view the file, if it is in a non-standard format or the file extension).  This might
% refer to a multi-page table or a figure.

% \subsection*{Additional file 2 --- Sample additional file title}
% Additional file descriptions text.


\end{bmcformat}
\end{document}








% LocalWords:  QuORUM HiTEC misassemblies Illumina Rhodobacter
% LocalWords:  contigs
