%% BioMed_Central_Tex_Template_v1.05
%%                                      %
%  bmc_article.tex            ver: 1.05 %
%                                       %


%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
%%                                     %%
%%  LaTeX template for BioMed Central  %%
%%     journal article submissions     %%
%%                                     %%
%%         <27 January 2006>           %%
%%                                     %%
%%                                     %%
%% Uses:                               %%
%% cite.sty, url.sty, bmc_article.cls  %%
%% ifthen.sty. multicol.sty            %%
%%                                     %%
%%                                     %%
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%


%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
%%                                                                 %%	
%% For instructions on how to fill out this Tex template           %%
%% document please refer to Readme.pdf and the instructions for    %%
%% authors page on the biomed central website                      %%
%% http://www.biomedcentral.com/info/authors/                      %%
%%                                                                 %%
%% Please do not use \input{...} to include other tex files.       %%
%% Submit your LaTeX manuscript as one .tex document.              %%
%%                                                                 %%
%% All additional figures and files should be attached             %%
%% separately and not embedded in the \TeX\ document itself.       %%
%%                                                                 %%
%% BioMed Central currently use the MikTex distribution of         %%
%% TeX for Windows) of TeX and LaTeX.  This is available from      %%
%% http://www.miktex.org                                           %%
%%                                                                 %%
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%


\NeedsTeXFormat{LaTeX2e}[1995/12/01]
\documentclass[10pt]{bmc_article}    



% Load packages
\usepackage{cite} % Make references as [1-4], not [1,2,3,4]
\usepackage{url}  % Formatting web addresses  
\usepackage{ifthen}  % Conditional 
\usepackage{multicol}   %Columns
\usepackage[utf8]{inputenc} %unicode support
%\usepackage[applemac]{inputenc} %applemac support if unicode package fails
%\usepackage[latin1]{inputenc} %UNIX support if unicode package fails
\usepackage{units}
\urlstyle{rm}
 
 
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%	
%%                                             %%
%%  If you wish to display your graphics for   %%
%%  your own use using includegraphic or       %%
%%  includegraphics, then comment out the      %%
%%  following two lines of code.               %%   
%%  NB: These line *must* be included when     %%
%%  submitting to BMC.                         %% 
%%  All figure files must be submitted as      %%
%%  separate graphics through the BMC          %%
%%  submission process, not included in the    %% 
%%  submitted article.                         %% 
%%                                             %%
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%                     


\def\includegraphic{}
\def\includegraphics{}



\setlength{\topmargin}{0.0cm}
\setlength{\textheight}{21.5cm}
\setlength{\oddsidemargin}{0cm} 
\setlength{\textwidth}{16.5cm}
\setlength{\columnsep}{0.6cm}

\newboolean{publ}

%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
%%                                              %%
%% You may change the following style settings  %%
%% Should you wish to format your article       %%
%% in a publication style for printing out and  %%
%% sharing with colleagues, but ensure that     %%
%% before submitting to BMC that the style is   %%
%% returned to the Review style setting.        %%
%%                                              %%
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
 

%Review style settings
%\newenvironment{bmcformat}{\begin{raggedright}\baselineskip20pt\sloppy\setboolean{publ}{false}}{\end{raggedright}\baselineskip20pt\sloppy}

%Publication style settings
\newenvironment{bmcformat}{\fussy\setboolean{publ}{true}}{\fussy}



% Begin ...
\begin{document}
\begin{bmcformat}


%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
%%                                          %%
%% Enter the title of your article here     %%
%%                                          %%
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%

\title{QuorUM: an error corrector for sequence generation sequencing reads}
 
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
%%                                          %%
%% Enter the authors here                   %%
%%                                          %%
%% Ensure \and is entered between all but   %%
%% the last two authors. This will be       %%
%% replaced by a comma in the final article %%
%%                                          %%
%% Ensure there are no trailing spaces at   %% 
%% the ends of the lines                    %%     	
%%                                          %%
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%


\author{Guillaume Mar\c{c}ais\correspondingauthor$^1$%
       \email{Guillaume Mar\c{c}ais\correspondingauthor - gmarcais@umd.edu}%
      \and
         Aleksey Zimin\correspondingauthor$^1$%
         \email{Aleksey Ziming\correspondingauthor- alekseyz@ipst.umd.edu}%
%       \and%
%         Neza Vodopivec$^1$%
%         \email{Neza Vodopivec - }
       and
         Jim Yorke$^1$%
         \email{Jim Yorke - yorke@umd.edu}
      }
      

%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
%%                                          %%
%% Enter the authors' addresses here        %%
%%                                          %%
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%

\address{%
    \iid(1)Institut for Physical Science and Technology, University of Maryland, College Park, MD
}%

\maketitle

%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
%%                                          %%
%% The Abstract begins here                 %%
%%                                          %%
%% The Section headings here are those for  %%
%% a Research article submitted to a        %%
%% BMC-Series journal.                      %%  
%%                                          %%
%% If your article is not of this type,     %%
%% then refer to the Instructions for       %%
%% authors on http://www.biomedcentral.com  %%
%% and change the section headings          %%
%% accordingly.                             %%   
%%                                          %%
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%


\begin{abstract}
  % Do not use inserted blank lines (ie \\) until main body of text.
\paragraph*{Background:}
Illumina Sequencing data can provide high coverage of a genome by relatively short ($\unit[100]{bp}$ to $\unit[150]{bp}$) reads at a low cost.
The base error rates in the reads vary greatly along the read sequence.
Low quality bases can be trimmed off or error-corrected.
Trimming, especially based on the quality scores, can eliminate large amounts of useful sequence in the reads.
Error correction is an alternative to trimming that takes advantage of the high coverage of the genome to make the high confidence base corrections in the reads.
Error correction allows for more effective use of the sequencing data thus reducing the coverage depth requirements in sequencing projects and increasing the quality of the resulting assemblies.
The vast majority of errors in Illumina data are substitutions, where a base call is incorrect.
Also, the sequence quality is usually higher on the 5’ end of a read and deteriorates toward the 3’ end of the read. Each base in the read has a quality score associated with it.
      
\paragraph*{Results:}
We propose a novel software, called QuorUM, for error correcting substitution error in second generation sequencing reads.
It provides accurate correction and is suitable for large data sets ($48$ billion bases corrected per hours on $48$ threads).

\paragraph*{Conclusions:} 
The QuorUM software is available under an open source license at \url{ftp://ftp.genome.umd.edu/pub/quorum}.

\end{abstract}



\ifthenelse{\boolean{publ}}{\begin{multicols}{2}}{}




%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
%%                                          %%
%% The Main Body begins here                %%
%%                                          %%
%% The Section headings here are those for  %%
%% a Research article submitted to a        %%
%% BMC-Series journal.                      %%  
%%                                          %%
%% If your article is not of this type,     %%
%% then refer to the instructions for       %%
%% authors on:                              %%
%% http://www.biomedcentral.com/info/authors%%
%% and change the section headings          %%
%% accordingly.                             %% 
%%                                          %%
%% See the Results and Discussion section   %%
%% for details on how to create sub-sections%%
%%                                          %%
%% use \cite{...} to cite references        %%
%%  \cite{koon} and                         %%
%%  \cite{oreg,khar,zvai,xjon,schn,pond}    %%
%%  \nocite{smith,marg,hunn,advi,koha,mouse}%%
%%                                          %%  
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%




%%%%%%%%%%%%%%%%
%% Background %%
%%
\section*{Background}
While second generation sequencing technologies have progressed tremendously and offer ever longer reads with low overall sequencing error rate ($\approx 1\%$), correcting errors in sequencing reads remains an important prepossessing step in \emph{de novo} genome assembly.
In general, error correcting the sequencing reads leads to assemblies with longer contiguous sequences and fewer misassemblies.




As the quality of base calling usually degrades toward the 3' ends of reads, an obvious error correction method is to trim all the reads on the 3' end, either by a fix amount or based on the quality values reported by the sequencing machine.
Although this simple trimming scheme will reduce the number of erroneous bases, it still leaves many errors in the reads and needlessly discard a lot of valid sequence.
Aggressive trimming results in fragmented assemblies.

On the other hand, trimming is an integral part of error correction.
The distribution of sequencing errors on the sequencing reads is complex and for some percentage of the reads, the sequence beyond a certain point contains too many errors to be corrected or, even worse, does not correspond to any sequence in the original genome.
It is important to trim those reads to avoid misassemblies.

% We need a paragraph or two on the various way to error correct: k-mer spectrum, alignment of reads, what else?

We propose a new error correction procedure and software package, named QuorUM (Quality Optimized Reads from the University of Maryland), that provides accurate trimming and error correction.
In addition, QuorUM is flexible: it can correct reads of varying length and reads containing ambiguous base calls (Ns).
It is also fast and can tackle the large data sets produced by today's high throughput sequencing machine.

We compare the error corrector by comparing the error corrected reads to the finished sequence of the organism.
We use the evaluation toolkit of~\cite{Yang2012} with some modifications (see~\ref{sec:Methods}).

 
%%%%%%%%%%%%%%%%%%%%%%%%%%%%
%% Results and Discussion %%
%%
\section*{Results and Discussion}

We evaluated the error correction software using multiple metrics.



    

%%%%%%%%%%%%%%%%%%%%%%
\section*{Conclusions}


  
%%%%%%%%%%%%%%%%%%
\section*{Methods}
\label{sec:Methods}

\subsection*{Error correction}

QuorUM uses the Jellyfish~\cite{Marcais2011} software to count the number of occurrences of the $k$-mers in the input reads twice.
First for all $k$-mers and second for the $k$-mers in which all bases have a quality value greater than a threshold (by default $5$ above the minimum quality).
Then, every read is examined in turn.
QuorUM searches for an \emph{anchor} $k$-mer: a $k$-mer whose number of occurrences is higher than a threshold $a$ ($a = 3$ by default). 
Starting from this anchor, QuorUM will look at every base, extending toward the 5' and 3' ends, and decide to make a correction based on the $k$-mer ending at that base.
Let $m = xb$ bet the $k$-mer, where $|m| = k$, $x$ is a prefix with $|x| = k - 1$, and $b$ is the base under consideration.
If $m$ has a number of occurrences greater than $a$, then no correction is made.
Otherwise, if there is a unique alternative $k$-mer obtained by substituting $b$ with another nucleotide $b'$ with count at least $3$ times higher, QuorUM substitute $b$ for $b'$ in the read.
In the case where no substitution is found, QuorUM trims the read at that position.
The maximum of correction made in a window of length $10$ is by default capped at $3$.
QuorUM will trim the read at the beginning of the window if the number of errors is too high.

The procedure uses the high-quality $k$-mer database in priority.
Precisely, if any of the occurrence counts for the four possible base $b$ is non-zero in the high quality database, then QuorUM uses the high quality database.
Otherwise, the database with all $k$-mers is used.
In effect, QuorUM will replace low quality $k$-mers by high quality $k$-mers when possible.

\subsection*{Evaluation}

We use the evaluation toolkit of Yang \emph{et al.}~\cite{Yang2012} with the following modifications.
First, we use Bowtie2~{\bf cite} in local mode, rather than BWA~{\bf cite} to align the reads to the finished sequence.
By default, BWA imposes that a read aligns on its entire length.
In other words, clipped sequences at the 3' and 5' ends decrease the alignment score.
Bowtie2 in local mode does not penalize for clipped sequences.
As the sequence quality degrades toward the 3' end, many reads containing significant amount of valid sequence would not align with BWA because of bad sequence on the 3' end.
But all these reads, after proper trimming, can be used for assembly, and therefore should be included for evaluating error correctors.

% Do we care about this?
Second, we report the gain, defined as $G = (TP - FP) / (TP + FN)$ and the false positive rate, defined $FPR = FP / (TP + FN)$.




    
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
\section*{Authors contributions}


    

%%%%%%%%%%%%%%%%%%%%%%%%%%%
\section*{Acknowledgements}
  \ifthenelse{\boolean{publ}}{\small}{}
  Text for this section \ldots


 
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
%%                  The Bibliography                       %%
%%                                                         %%              
%%  Bmc_article.bst  will be used to                       %%
%%  create a .BBL file for submission, which includes      %%
%%  XML structured for BMC.                                %%
%%                                                         %%
%%                                                         %%
%%  Note that the displayed Bibliography will not          %% 
%%  necessarily be rendered by Latex exactly as specified  %%
%%  in the online Instructions for Authors.                %% 
%%                                                         %%
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%


{
  \ifthenelse{\boolean{publ}}{\footnotesize}{\small}
  \bibliographystyle{bmc_article}  % Style BST file
  \bibliography{bmc_article}
}     % Bibliography file (usually '*.bib' ) 

%%%%%%%%%%%

\ifthenelse{\boolean{publ}}{\end{multicols}}{}

%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
%%                               %%
%% Figures                       %%
%%                               %%
%% NB: this is for captions and  %%
%% Titles. All graphics must be  %%
%% submitted separately and NOT  %%
%% included in the Tex document  %%
%%                               %%
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%

%%
%% Do not use \listoffigures as most will included as separate files

\section*{Figures}
\subsection*{Figure 1 - Sample figure title}
A short description of the figure content
should go here.

\subsection*{Figure 2 - Sample figure title}
Figure legend text.



%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
%%                               %%
%% Tables                        %%
%%                               %%
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%

%% Use of \listoftables is discouraged.
%%
\section*{Tables}
\subsection*{Table 1 - Sample table title}
Here is an example of a \emph{small} table in \LaTeX\ using  
\verb|\tabular{...}|. This is where the description of the table 
should go. \par \mbox{}
\par
\mbox{
  \begin{tabular}{|c|c|c|}
    \hline \multicolumn{3}{|c|}{My Table}\\ \hline
    A1 & B2  & C3 \\ \hline
    A2 & ... & .. \\ \hline
    A3 & ..  & .  \\ \hline
  \end{tabular}
}
\subsection*{Table 2 - Sample table title}
Large tables are attached as separate files but should
still be described here.



%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
%%                               %%
%% Additional Files              %%
%%                               %%
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%

\section*{Additional Files}
\subsection*{Additional file 1 --- Sample additional file title}
Additional file descriptions text (including details of how to
view the file, if it is in a non-standard format or the file extension).  This might
refer to a multi-page table or a figure.

\subsection*{Additional file 2 --- Sample additional file title}
Additional file descriptions text.


\end{bmcformat}
\end{document}







